diff --git a/libcaf_core/CMakeLists.txt b/libcaf_core/CMakeLists.txt
index 15ef0e0..807e0a0 100644
--- a/libcaf_core/CMakeLists.txt
+++ b/libcaf_core/CMakeLists.txt
@@ -140,45 +140,53 @@ list(APPEND LIBCAF_CORE_HDRS
   "${CMAKE_CURRENT_BINARY_DIR}/caf/detail/build_config.hpp"
 )
 
-add_custom_target(libcaf_core)
 
-# build shared library if not compiling static only
-if (NOT CAF_BUILD_STATIC_ONLY)
-  add_library(libcaf_core_shared SHARED ${LIBCAF_CORE_SRCS} ${LIBCAF_CORE_HDRS})
-  target_link_libraries(libcaf_core_shared ${CAF_EXTRA_LDFLAGS})
-  target_include_directories(libcaf_core_shared PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:include>
-  )
-  set_target_properties(libcaf_core_shared
-    PROPERTIES
+add_library(core ${LIBCAF_CORE_SRCS} ${LIBCAF_CORE_HDRS})
+
+target_include_directories(core PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<INSTALL_INTERFACE:include>
+)
+
+
+target_link_libraries(core PUBLIC ${CAF_EXTRA_LDFLAGS})
+
+target_compile_features(core PUBLIC cxx_std_11)
+
+set_target_properties(core
+  PROPERTIES
     SOVERSION ${CAF_VERSION}
     VERSION ${CAF_VERSION}
     OUTPUT_NAME caf_core
-  )
-  install(TARGETS libcaf_core_shared
-    RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-  )
-  add_dependencies(libcaf_core_shared libcaf_core)
-endif ()
+)
+
 
-# build static library only if --build-static or --build-static-only was set
-if (CAF_BUILD_STATIC_ONLY OR CAF_BUILD_STATIC)
-  add_library(libcaf_core_static STATIC ${LIBCAF_CORE_HDRS} ${LIBCAF_CORE_SRCS})
-  target_link_libraries(libcaf_core_static ${CAF_EXTRA_LDFLAGS})
-  target_include_directories(libcaf_core_static PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:include>
-  )
-  set_target_properties(libcaf_core_static PROPERTIES OUTPUT_NAME caf_core_static)
-  install(TARGETS libcaf_core_static ARCHIVE DESTINATION lib)
-  add_dependencies(libcaf_core_static libcaf_core)
-endif ()
+###############################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################
+# Install
+###############################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################
+include(GNUInstallDirs)
 
-install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/caf"
-        DESTINATION include
-        FILES_MATCHING PATTERN "*.hpp"
+install(
+  TARGETS
+    core
+  EXPORT
+    caf-core-targets
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+
+install(
+  EXPORT 
+    caf-core-targets
+  FILE
+    caf-core-targets.cmake
+  NAMESPACE 
+    Caf::
+  DESTINATION
+    ${CMAKE_INSTALL_DATAROOTDIR}/caf
+)
+  
+install(DIRECTORY  "${CMAKE_CURRENT_SOURCE_DIR}/caf"
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  FILES_MATCHING PATTERN "*.hpp"
 )
